### ISSUE TITLE
SysmonConfigPusher won't start

### ISSUE THREAD

**Issue by reswob10** [LaresLLC/SysmonConfigPusher#3]
I have a Windows server 2019 VM running on proxmox with 8 core and 8GB RAM.  
I installed dotnet-sdk-5.0.205-win-x64.exe and rebooted and the registry shows 
![Screenshot 2021-07-14 182930](https://user-images.githubusercontent.com/1158116/125701199-6cd5ae84-7afb-49ad-bf3b-53d10eb7da33.png)

I double clicked on SysmonConfigPusher.exe but it wouldn't start.  I got the following errors in the Application Log:

Application Log

Event 1001:
-------------
Fault bucket 2234889022483767748, type 5
Event Name: CLR20r3
Response: Not available
Cab Id: 0

Problem signature:
P1: SysmonConfigPusher.exe
P2: 1.0.0.0
P3: 9cab404c
P4: PresentationFramework
P5: 4.7.3781.0
P6: 60230bf3
P7: 2262
P8: f6
P9: System.Windows.Markup.XamlParse
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERCDB0.tmp.dmp
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERCEBA.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERCF0A.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERCF08.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERCF18.tmp.txt

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_SysmonConfigPush_52fd11c4a5c278e9eb616db83eb574c9e6afc8b_0c91de3f_1b18d1d6

Analysis symbol: 
Rechecking for solution: 0
Report Id: 1ab7b977-73e4-431c-8602-c7740a8a344c
Report Status: 268435456
Hashed bucket: 1229985a875f32ef7f03ebb99990a1c4
Cab Guid: 0

Event 1000
--------------
Faulting application name: SysmonConfigPusher.exe, version: 1.0.0.0, time stamp: 0x9cab404c
Faulting module name: KERNELBASE.dll, version: 10.0.17763.1879, time stamp: 0x55fe51b0
Exception code: 0xe0434352
Fault offset: 0x00122802
Faulting process id: 0x1a80
Faulting application start time: 0x01d778fe511d26e5
Faulting application path: C:\tools\SysmonConfigPusher\SysmonConfigPusher-main\bin\Release\SysmonConfigPusher.exe
Faulting module path: C:\Windows\System32\KERNELBASE.dll
Report Id: 1ab7b977-73e4-431c-8602-c7740a8a344c
Faulting package full name: 
Faulting package-relative application ID: 

Event 1026
----------
Application: SysmonConfigPusher.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.IO.FileNotFoundException
   at SysmonConfigPusher.MainWindow..ctor()

Exception Info: System.Windows.Markup.XamlParseException
   at System.Windows.Markup.WpfXamlLoader.Load(System.Xaml.XamlReader, System.Xaml.IXamlObjectWriterFactory, Boolean, System.Object, System.Xaml.XamlObjectWriterSettings, System.Uri)
   at System.Windows.Markup.WpfXamlLoader.LoadBaml(System.Xaml.XamlReader, Boolean, System.Object, System.Xaml.Permissions.XamlAccessLevel, System.Uri)
   at System.Windows.Markup.XamlReader.LoadBaml(System.IO.Stream, System.Windows.Markup.ParserContext, System.Object, Boolean)
   at System.Windows.Application.LoadBamlStreamWithSyncInfo(System.IO.Stream, System.Windows.Markup.ParserContext)
   at System.Windows.Application.LoadComponent(System.Uri, Boolean)
   at System.Windows.Application.DoStartup()
   at System.Windows.Application.<.ctor>b__1_0(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(System.Object)
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(System.Windows.Interop.MSG ByRef)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(System.Windows.Threading.DispatcherFrame)
   at System.Windows.Threading.Dispatcher.PushFrame(System.Windows.Threading.DispatcherFrame)
   at System.Windows.Application.RunDispatcher(System.Object)
   at System.Windows.Application.RunInternal(System.Windows.Window)
   at System.Windows.Application.Run(System.Windows.Window)
   at SysmonConfigPusher.App.Main()

Any idea what I'm missing or not doing?

**Comment by Antonlovesdnb** [comment]
Hey thanks very much for the detailed info, it looks like it's expecting a file that isn't available, is the SysmonConfigPusher.exe file being ran from the same folder as the DLLs from the zip file in the releases section? If that's not the issue I can definitely take a close look 👍

**Comment by reswob10** [comment]
I will try this tonight...  Thanks

Craig L Bowser
____________________________

This email is measured by size.  Bits and bytes may have settled during
transport.

**Comment by reswob10** [comment]
These are the files in C:\tools\SysmonConfigPusher\SysmonConfigPusher-main\bin\Release

Ceen.Common.dll
Ceen.Httpd.dll
Microsoft.Owin.Host.HttpListener.dll
SysmonConfigPusher.exe
SysmonConfigPusher.exe.config


Should I move the files to a different directory and run them from there?
Cause it works from the C:\tools\SysmonConfigPusher\SysmonConfigPusher-main\bin\Debug folder....

**Comment by Antonlovesdnb** [comment]
Try grabbing the zip from the releases section here: 

https://github.com/LaresLLC/SysmonConfigPusher/files/6626439/SysmonConfigPusher.zip

once you unzip the file you should see 

Ceen.Common.dll
Ceen.Httpd.dll
Microsoft.Owin.Host.HttpListener.dll
Serilog.dll
Serilog.Sinks.File.dll
SysmonConfigPusher.exe
SysmonConfigPusher.exe.config

Looks like when you build it, it doesn't copy the Serilog files that are needed - I'll look into building an "all in one" type exe - let me know if that works for you

**Comment by reswob10** [comment]
Yup and it ran by double clicking!

Thanks

Craig L Bowser
____________________________

This email is measured by size.  Bits and bytes may have settled during
transport.

**Comment by Antonlovesdnb** [comment]
Awesome glad it worked for you - let me know if you run into any other issues :)

**Comment by reswob10** [comment]
errata:

For the requirement: Endpoints must be able to reach the host that
SysmonConfigPusher is running on, over port 80---> make sure traffic
allowed in via Window Firewall to sysmonconfigpusher.exe on port 80

If you already installed sysmon and then you decided to use this program,
you STILL need to perform the step to download the newest sysmon.exe.  I
know that sounds silly, but the program relies on sysmon.exe being in the
C:\SysmonFiles directory and updating the config won't work if it's not
there.  (doesn't throw a good error saying something like "Can't find
sysmon.exe" either....)

IOW, I had pushed a configuration update to the client machines.  The
update was the only thing in C:\SysmonFiles directory.  When I tried to
Update the config, nothing happened and I didn't see Sysmon EventID 16.
Once I clicked Push Newest Executable from Sysinternals which put
sysmon.exe in the target directory, the update worked.

Craig L Bowser
____________________________

This email is measured by size.  Bits and bytes may have settled during
transport.
